### ISSUE TITLE
t-model with t-on-input seems to misbehave

### ISSUE THREAD

**Issue by SimonGenin** [odoo/owl#700]
I'm playing around with owl. 
I have a simple component that encapsulates an input field. I'm trying to bubble up an event on the input event (the simplest way, no store involved). Using both t-model and t-on-input makes for unwanted behavior.

```ts
import { Component, tags, useState } from "@odoo/owl";
import SpecialField from "./SpecialField"

const { xml, css } = tags;

const TEMPLATE = xml/* xml */`
   <div>
        <SpecialField t-on-value-changed="updateValue"/>
        <p t-esc="state.text"></p>   
   </div>
`

export class App extends Component {

    static template = TEMPLATE;

    static components = { SpecialField }

    state = useState({ text: "" })

    updateValue({detail}) {
        this.state.text = detail;
    }

}
```

```ts
import { Component, tags, hooks, useState } from "@odoo/owl"

const { xml } = tags;

const TEMPLATE = xml/* xml */`
    <input type="text" t-on-input="changing" t-model="state.text" t-on-blur="clear"/>
`

export default class SpecialField extends Component {

    static template = TEMPLATE

    state = useState({ text: "" })

    clear() {
        this.state.text = "";
    }

    changing() {
        this.trigger('value-changed', this.state.text)
    }

}
```

With the `t-on-input="changing"`, `this.state.text` is always `""` and reactivity is lost.  `t-on-input.stop` doesn't change anything. 

As I am new to owl, is it expected behavior ? Am I doing something wrong ? And if t-on-input shouldn't be use with t-model, shouldn't we prevent its use at the framework level ? 

Also, to avoid t-on-input and t-model on the same element, is there a way to observe the state update ? Something like: `this.state.text.onchange( function (event) { ... } )`. I couldn't find where this could be mentioned.

**Comment by ged-odoo** [comment]
You're right, it looks like a bug. I assume that it's because internally `t-model` add a listener on 'input' event, just like `t-on-input`

**Comment by ged-odoo** [comment]
About your second question, I thought about a `watch` feature, but the correct API is not obvious, and so, I decided not to implement it for now.  We try to keep owl minimalist if in doubt.  But in my work on owl 2.0, I reimplemented the reactivity system, and i think I may have a design for this issue.

**Comment by SimonGenin** [comment]
@ged-odoo I tried a few things but it didn't work. I might need some explanation on how the extra handlers work in owl to wrap my head around it. 
The solution I tried was to check if an 'input' handler already exist and if it did, make the handler such as: 
```js 
newInputHandler = (ev) => { InputhandlerThatWasThereBefore(); newInputHandler() };
```
So it wouldn't erase the other one. But I don't know, it brought plenty of problems and the code was getting uglier. 

We could also forbid the use of t-model when t-on-input is already used. And throw an error. That kind of limit the user but that's a solution. By the way, I would forbid the t-model instead of the t-on-input because the t-model behaviour can be recreate with the t-on-input, not the opposite.
